src/plugins/intel_gpu/src/graph/border.cpp

```diff
@@ -65,6 +65,10 @@ std::vector<layout> border_inst::calc_output_layouts(border_node const& /*node*/
     };
 
     auto& memory_deps = impl_param.memory_deps;
+    if (desc->pad_value == 0 && (is_begin_mem || is_end_mem) && memory_deps.empty()) {
+        return {layout{ShapeType::dynamic(static_cast<int64_t>(input_shapes[0].size())), input0_layout.data_type, input0_layout.format}};
+    }
+
```

Comment:
[random spot] Could you add shape infer test for border?

Reply:
Also please have a look at the smoke_GPUPadDynamic2D/PadLayerGPUTest.CompareWithRefs/IS=[?.?]_TS=(5.36)_(3.16)_Prc=f32_padsBegin=(0.1)_padsEnd=(1.2)_constantInput=CONSTANT/PARAMETER/CONSTANT_PadMode=edge test. With PR https://github.com/openvinotoolkit/openvino/pull/18925 this (and some other) test fails due to nullptr passed to make_tensor. So I assume that the shape infer impl has logical error, thus please ensure that such case is covered by shape infer unit tests

Reply:
> [random spot] Could you add shape infer test for border?

Yes, the shape infer test is added. **pad_test_params** in the test is used, and I'm not sure if it is proper.
Could you review the test please?

Reply:
I think the case that I mentioned above is still not covered and not fixed. I think it can be covered to some degree by changing border type for `pad_test_non_constant_input_end` test case to `EDGE` value. Another uncovered thing is the case when `value` input has `Constant` type in ngraph, but mode != Constant. In that case `data` primitive is created for that argument and it may behave a little bit differently comparing to `input_layout`